src/snowflake/snowpark/_internal/telemetry.py

```diff
@@ -168,6 +168,10 @@ def wrap(*args, **kwargs):
         ]._session.sql_simplifier_enabled
         try:
             api_calls[0][TelemetryField.QUERY_PLAN_HEIGHT.value] = plan.plan_height
+            # The uuid for df._select_statement can be different from df._plan. We always
+            # use plan uuid to track the queries.
+            uuid = args[0]._plan.uuid
+            api_calls[0][CompilationStageTelemetryField.PLAN_UUID.value] = uuid
```

Comment:
once we add this, does this mean all api telemetry will have plan id associated with it?

Reply:
this applies to each collect telemetry. We send a list of api calls associated with a collect. In this section, we are modifying the first telemetry call to associate the plan uuid.

Reply:
you mentioned "each collect telemetry" and "api calls", and "telemetry call". I now we call send_** to send the telemetry, and after the number of telemetry is buffered, or session closed, the telemetries will be sent over. for example, if the buffer have [send_post_compilation_stage_telemetry1, send_post_compilation_stage_telemetry2, send_post_compilation_stage_telemetry3], will each of those have the plan_id associated?

Reply:
I want to make 2 clarifications.

1. We are adding plan id to `df_collect_api_telemetry`, not every `send_*` telemetry call.
2. The telemetry data sent by `df_collect_api_telemetry` for a dataframe that is construct using `session.table().select()` currently looks like so
```
{
  "api_calls": [
    {
      "name": "Session.table",
      "query_plan_complexity": {"..."},
      "query_plan_height": 8,
      "query_plan_num_duplicate_nodes": 1,
      "sql_simplifier_enabled": true
    },
    {
      "name": "DataFrame.select"
    },
  ]
...
}
```

after this change it will look like so:
```
{
  "api_calls": [
    {
      "name": "Session.table",
      "plan_uuid": "<uuid>",
      "query_plan_complexity": {"..."},
      "query_plan_height": 8,
      "query_plan_num_duplicate_nodes": 1,
      "sql_simplifier_enabled": true
    },
    {
      "name": "DataFrame.select"
    },
  ]
...
}
```

Reply:
I see. Thanks a lot for the explaination